lib/index.ts

```diff
@@ -0,0 +1,25 @@
+import * as nodeStore from './stores/node-storage';
+import { getStorage as $getStorage } from './storage';
+import type { BalenaSettingsStorage, StorageLike } from './types';
+
+export type { BalenaSettingsStorage } from './types';
+
+export const getStorage = (options: {
+	dataDirectory?: string;
+}): BalenaSettingsStorage => {
+	let store: StorageLike;
+	if (typeof window !== 'undefined') {
+		// use dynamic imports so that node apps have less files to read on startup.
+		const localStore =
+			require('./stores/local-storage') as typeof import('./stores/local-storage');
+		store = localStorage.isSupported()
+			? localStore.createStorage(options?.dataDirectory)
+			: (
+					require('./stores/virtual-storage') as typeof import('./stores/virtual-storage')
+			  ).createStore(options?.dataDirectory);
```

Comment:
@thgreasi Why are we passing dataDIrectory for the virtual-storage? Shouldn't it always be empty?

Reply:
I guess the same would be valid for localStorage? Not exactly sure.

More important, what do you think of having at some point in here a test to check if `options?.dataDirectory === ':memory:` so that we can run in-memory in the nodejs version but without isBrowser being true? this way we can run in memory and also not block some functionalities that are not allowed in browser

Reply:
I'm only passing it for consistency reasons on how we call the common interface @otaviojacobi , we can for sure remove it.
The reason I started this PR actually, and splitted virtual storage out of the local-storage.ts was your suggestion about supporting an explicit way of targeting a memory storage. But better do that in a follow-up minor.